datumaro/plugins/datumaro_format/converter.py

```diff
@@ -383,3 +386,41 @@ def patch(cls, dataset, patch, save_dir, **kwargs):
             )
             if osp.isdir(related_images_path):
                 shutil.rmtree(related_images_path)
+
+    @staticmethod
+    def _filterout_for_single_selection(item, categories):
+        if AnnotationType.label not in categories:
+            return item
+
+        name2parent_ss = {}
+        for label_category in categories[AnnotationType.label]:
+            name2parent_ss[label_category.name] = (
+                label_category.parent,
+                label_category.single_selection,
+            )
+
+        # collect childrens that have the same parent which only allow single-selection
+        parent2ss_indices = {}
+        for i, annotation in enumerate(item.annotations):
+            if annotation._type != AnnotationType.label:
+                continue
+
+            label_name = categories[AnnotationType.label][annotation.label].name
+
+            parent, _ = name2parent_ss[label_name]
+            if len(parent) == 0:
+                continue
+
+            _, single_selection = name2parent_ss[parent]
+
+            if single_selection:
+                parent2ss_indices[parent] = parent2ss_indices.get(parent, []) + [i]
```

Comment:
If you use `parent2ss_indices = defaultdict(list)`, then this line will be `parent2ss_indices[parent] += [i]`.

Reply:
cool!